src/osemgrep/language_server/Test_LS_e2e.ml

```diff
@@ -234,28 +240,30 @@ let receive_request (info : info) : Request.t Lwt.t =
 (*****************************************************************************)
 
 let git_tmp_path () =
-  Testutil_files.with_tempdir ~persist:true (fun dir ->
-      let dir = Fpath.to_string dir in
-      (* I don't know why, but the tests will hang in OCaml if we do
-         not chdir here.
-      *)
-      if not !Common.jsoo then Sys.chdir dir;
-      checked_command (String.concat " " [ "git"; "-C"; dir; "init" ]);
+  let _orig_dir = Sys.getcwd () in
+  Testutil_files.with_tempdir ~persist:true (*~chdir:true*) (fun dir ->
+      (* TODO: investigate/report
+         Mysterious bug: the test will hang as we chdir back into the original
+         directory. *)
+      if not !Common.jsoo then Sys.chdir !!dir;
+      checked_command (String.concat " " [ "git"; "-C"; !!dir; "init" ]);
       checked_command
         (String.concat " "
            [
              "git";
              "-C";
-             dir;
+             !!dir;
              "config";
              "user.email";
              "baselinetest@semgrep.com";
            ]);
       checked_command
         (String.concat " "
-           [ "git"; "-C"; dir; "config"; "user.name"; "Baseline Test" ]);
+           [ "git"; "-C"; !!dir; "config"; "user.name"; "Baseline Test" ]);
       checked_command
-        (String.concat " " [ "git"; "-C"; dir; "checkout"; "-B"; "main" ]);
+        (String.concat " " [ "git"; "-C"; !!dir; "checkout"; "-B"; "main" ]);
+      (* !!!!!!!!!!!!!!! This call causes hanging !!!!!!!!!!!!!!! *)
+      (* Sys.chdir _orig_dir; *)
       dir)
```

Comment:
This is a nasty bug. Could be a bad interaction between some system calls and lwt, I don't know.